src/components/loginComponents/PostStoryForm.js

```diff
@@ -13,41 +15,73 @@ export default function PostStoryForm({
-      hyperlink,
-      story_title,
-      place_ids: [place_id],
-      date: Date().toString(),
-    });
-    fetch('https://backend-mongo-stories.azurewebsites.net/stories/create', {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({
-        user_id: user.userId,
-        hyperlink,
-        story_title,
-        place_ids: [place_id],
-        date: Date().toString(),
-      }),
-    }).then(() => {
+  async function handleSubmit(hyperlink, preview, place_id) {
+    fetch(
+      'https://climatetree-api-gateway.azurewebsites.net/stories/create',
+      {
+        method: 'POST',
+        headers: {
+          Authorization: "Bearer " + user.jwt,
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          user_id: user.userId,
+          posted_by: user.username,
+          hyperlink,
+          story_title: preview.title || hyperlink,
+          description: preview.description || hyperlink,
+          image: preview.image || '',
+          place_ids: [place_id],
+          date: Date().toString(),
+          strategy: strategies.length ? strategies : ['Other'],
+          sector: sector.length ? [sector] : ['Other'],
+          solution: solution.length ? [solution] : ['Other'],
+        }),
+      }
+    ).then(() => {
       setMyStories([
         ...myStories,
         {
           user_id: user.userId,
+          posted_by: user.username,
           hyperlink,
-          story_title,
+          story_title: preview.title || hyperlink,
+          description: preview.description || hyperlink,
+          image: preview.image || '',
+          place_ids: [place_id],
           date: Date().toString(),
+          strategy: strategies.length ? strategies : ['Other'],
+          sector: sector.length ? [sector] : ['Other'],
+          solution: solution.length ? [solution] : ['Other'],
         }
       ]);
     }).catch((error) => {
       console.log(error);
     });
     closeForm();
   }
+  const [strategies, setStrategies] = useState([]);
+  const toggleStrategy = (strategyName) => {
+    if (strategies.includes(strategyName)) {
+      setStrategies(strategies.filter(s => s !== strategyName));
+    } else {
+      setStrategies([...strategies, strategyName]);
+    }
+  }
+  const [sector, setSector] = useState('');
+  const [allSectors, setAllSectors] = useState([]);
+  const [solution, setSolution] = useState('');
+  const [allSolutions, setAllSolutions] = useState([]);
+  useEffect(() => {
+    (async () => {
+      const solutionResponse = await fetch('https://climatetree-api-gateway.azurewebsites.net/stories/all/solution');
+      const solutions = await solutionResponse.json();
+      setAllSolutions(solutions);
+      const sectorResponse = await fetch('https://climatetree-api-gateway.azurewebsites.net/stories/all/sector');
+      const sectors = await sectorResponse.json();
+      setAllSectors(sectors);
```

Comment:
For the future we should think about returning all solutions and sectors in the same callout